### PR TITLE
improve max_valid_n and doc

### PR DESCRIPTION
**Pull request by sonichi**
## Why are these changes needed?

* Improve the bound of max_valid_n for better initial validity check.
* Improve the documentation and installation options to reflect the support of LLMs.

## Checks

- [x] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [x] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

File: website/docs/Getting-Started.md [hunk 1]
```diff
@@ -3,17 +3,17 @@
...
+1. For common machine learning or AI tasks like classification, regression and generation, it quickly finds quality models for user-provided data with low computational resources. It supports both classical machine learning models and deep neural networks, including large language models.
 
 2. It is easy to customize or extend. Users can find their desired customizability from a smooth range: minimal customization (computational resource budget), medium customization (e.g., scikit-style learner, search space and metric), or full customization (arbitrary training and evaluation code). Users can customize only when and what they need to, and leave the rest to the library.
 
 3. It supports fast and economical automatic tuning, capable of handling large search space with heterogeneous evaluation cost and complex constraints/guidance/early stopping. FLAML is powered by a new, [cost-effective
 hyperparameter optimization](Use-Cases/Tune-User-Defined-Function#hyperparameter-optimization-algorithm)
-and learner selection method invented by Microsoft Research.
+and model selection method invented by Microsoft Research, and many followup [research studies](Research).
```
Comment (thinkall) [inline website/docs/Getting-Started.md:16]:
```suggestion
and model selection method invented by Microsoft Research, and many followup [research studies](https://microsoft.github.io/FLAML/docs/Research).
```

**Review by thinkall — COMMENTED** [review]
Could you also update the version in automl_nlp.ipynb? Is it necessary to fix it to `1.0.11`?

**Comment by sonichi** [comment]
> Could you also update the version in automl_nlp.ipynb? Is it necessary to fix it to `1.0.11`?

@liususan091219 Could you please answer this question? Thanks.

**Comment by liususan091219** [comment]
> Could you also update the version in automl_nlp.ipynb? Is it necessary to fix it to `1.0.11`?

Thanks for the question. It is not necessary to fix to 1.0.11. In fact, the nlp notebook in the main branch is out dated. The up-to-date notebook can be found here: https://github.com/microsoft/FLAML/blob/tutorial-aaai23/notebook/automl_nlp.ipynb. How about creating another PR to update the main branch nlp notebook? @sonichi

**Comment by sonichi** [comment]
> > Could you also update the version in automl_nlp.ipynb? Is it necessary to fix it to `1.0.11`?
> 
> Thanks for the question. It is not necessary to fix to 1.0.11. In fact, the nlp notebook in the main branch is out dated. The up-to-date notebook can be found here: https://github.com/microsoft/FLAML/blob/tutorial-aaai23/notebook/automl_nlp.ipynb. How about creating another PR to update the main branch nlp notebook? @sonichi

Yes please.

File: README.md [hunk 2]
```diff
@@ -21,13 +21,13 @@
...
 FLAML is a lightweight Python library that finds accurate machine
 learning models automatically, efficiently and economically. It frees users from selecting
-learners and hyperparameters for each learner. It can also be used to tune generic hyperparameters for MLOps workflows, pipelines, mathematical/statistical models, algorithms, computing experiments, software configurations and so on.
+models and hyperparameters for each model. It can also be used to tune generic hyperparameters for large language models (LLM), MLOps/LMOps workflows, pipelines, mathematical/statistical models, algorithms, computing experiments, software configurations and so on.
 
-1. For common machine learning tasks like classification and regression, it quickly finds quality models for user-provided data with low computational resources. It supports both classifcal machine learning models and deep neural networks.
+1. For common machine learning or AI tasks like classification, regression and generation, it quickly finds quality models for user-provided data with low computational resources. It supports both classical machine learning models and deep neural networks, including large language models.
```
Comment (liususan091219) [inline README.md:26]:
```suggestion
1. For common machine learning or AI tasks like classification, regression, and text generation, it quickly finds quality models for user-provided data with low computational resources. It supports both classical machine learning models and deep neural networks, including large language models such as the OpenAI GPT-3 models.
```

File: flaml/integrations/oai/completion.py [hunk 3]
```diff
@@ -187,34 +188,40 @@ def eval(cls, config: dict, prune=True, eval_only=False):
         prune_hp = cls._prune_hp
         metric = cls._metric
         config_n = config[prune_hp]
-        max_tokens = config["max_tokens"]
+        max_tokens = config.get("max_tokens", 16)  # default value is 16
```
Comment (liususan091219) [inline flaml/integrations/oai/completion.py:191]: why 16? is there any reference for how 16 is selected?

File: flaml/integrations/oai/completion.py [hunk 4]
```diff
@@ -187,34 +188,40 @@ def eval(cls, config: dict, prune=True, eval_only=False):
...
-            min_invalid_n = cls._get_min_invalid_n(region_key, max_tokens)
-            if min_invalid_n is not None and config_n >= min_invalid_n:
-                if config_n > max_valid_n:
+            if cls.avg_input_tokens:
+                # max_tokens bounds the maximum tokens
+                # so using it we can calculate a valid n according to the avg # input tokens
+                max_valid_n = max(
+                    max_valid_n,
+                    int((target_n_tokens - cls.avg_input_tokens) // max_tokens),
+                )
```
Comment (liususan091219) [inline flaml/integrations/oai/completion.py:203]: One question: this does not need to be applied to min_invalid_n?

File: website/docs/Getting-Started.md [hunk 5]
```diff
@@ -3,17 +3,17 @@
 <!-- ### Welcome to FLAML, a Fast Library for Automated Machine Learning & Tuning! -->
 
 FLAML is a lightweight Python library that finds accurate machine
-learning models automatically, efficiently and economically. It frees users from selecting learners and hyperparameters for each learner.
+learning models automatically, efficiently and economically. It frees users from selecting models and hyperparameters for each model.
 
 ### Main Features
 
-1. For common machine learning tasks like classification and regression, it quickly finds quality models for user-provided data with low computational resources. It supports both classical machine learning models and deep neural networks.
+1. For common machine learning or AI tasks like classification, regression and generation, it quickly finds quality models for user-provided data with low computational resources. It supports both classical machine learning models and deep neural networks, including large language models.
```
Comment (liususan091219) [inline website/docs/Getting-Started.md:10]:
```suggestion
1. For common machine learning or AI tasks like classification, regression, and text generation, it quickly finds quality models for user-provided data with low computational resources. It supports both classical machine learning models and deep neural networks, including large language models such as the OpenAI GPT-3 model.
```

**Review by liususan091219 — CHANGES_REQUESTED** [review]

File: flaml/integrations/oai/completion.py (hunk 3 above)
Comment (sonichi) [reply flaml/integrations/oai/completion.py:191]: default value in OpenAI.

File: flaml/integrations/oai/completion.py (hunk 4 above)
Comment (sonichi) [reply flaml/integrations/oai/completion.py:203]: Right. max_tokens is only the upper limit.

File: flaml/integrations/oai/completion.py (hunk 3 above)
Comment (liususan091219) [reply flaml/integrations/oai/completion.py:191]: Okay. Shall we mention it's the default value in OpenAI?

**Review by liususan091219 — APPROVED** [review]

**Review by qingyun-wu — APPROVED** [review]

**Pull request merged**